templates/restoreTable.html

```diff
@@ -35,15 +35,74 @@ <h2>Restore table</h2>
 </script>
 
 <form name="restoreTableForm" onsubmit="return OnSubmitForm();">
-  Source project id<span style="color: red;">*</span>:<br><input type="text" name="projectId" value="" required><br>
-  Source dataset id<span style="color: red;">*</span>:<br><input type="text" name="datasetId" value="" required><br>
-  Source table id<span style="color: red;">*</span>:<br><input type="text" name="tableId" value="" required><br>
-  Source table partition id:<br><input type="number" min="9999999" max="100000000"
-                                       name="partitionId" value=""><br>
-  Target dataset id:<br><input type="text" name="targetDatasetId" value=""><br>
-  Restoration date:<br><input type="date" name="restorationDate" value=""><br>
-  <br>
-  <input type="submit" value="Submit">
+  <div class="form-group row">
+    <label for="sourceProjectId" class="col-sm-2 col-form-label">Source project id</label>
+    <div class="col-sm-8">
+      <input type="text" class="form-control" id="sourceProjectId" name="projectId"
+             placeholder="project id" required/>
+      <small id="sourceProjectIdHelp" class="form-text text-muted">Project id where dataset is
+        placed originally
+      </small>
+    </div>
+  </div>
+  <div class="form-group row">
+    <label for="sourceDatasetId" class="col-sm-2 col-form-label">Source dataset id</label>
+    <div class="col-sm-8">
+      <input type="text" class="form-control" id="sourceDatasetId" name="datasetId"
+             placeholder="dataset id">
+      <small id="sourceDatasetIdHelp" class="form-text text-muted">Dataset id where tables were
+        placed originally
+      </small>
+    </div>
+  </div>
+  <div class="form-group row">
+    <label for="sourceTableId" class="col-sm-2 col-form-label">Source table id</label>
+    <div class="col-sm-8">
+      <input type="text" class="form-control" id="sourceTableId" name="tableId"
+             placeholder="table id">
+      <small id="sourceTableIdHelp" class="form-text text-muted">Table id of original table</small>
+    </div>
+  </div>
+  <div class="form-group row">
+    <label for="sourcePartitionId" class="col-sm-2 col-form-label">Source partition id</label>
+    <div class="col-sm-8">
+      <input type="number" class="form-control" id="sourcePartitionId" name="partitionId"
+             placeholder="partition id">
+      <small id="sourcePartitionIdHelp" class="form-text text-muted">Optional. Partition id of
+        original table. If specified, only this single partition will be restored
```

Comment:
it's confusing -> if table is partitioned, then the only way to restore using that endpoint is to provide partition (in that case it is required) and restore each partition in that way ( if table is partitioned and partitionId won't be set then it wouldn't find backup in datastore). 

partitionId is set as optional only because non-partitioned doesn't need that parameter

it would be nice to emphasise that in that description.

Reply:
Fixed